lib/wadler/print.rb

```diff
@@ -19,7 +24,7 @@ def initialize(margin: 80, new_line: "\n")
     # @return [String]
     def output
       tokens << Oppen.eof
```

Comment:
This should be conditioned on whether the last token was indeed a eof.

Reply:
The user cannot add an eof if he is using the methods from Wadler, so this case would never happen if the library was used as intended

Reply:
In ruby, everything could be _opened_ (pun intended) and modified, even the classes defined in external libs.

We might also officially allow users to inject tokens from the Wadler interface in the future.

So this safety measure costs us nothing, and buys 1 less bugfix/release.